pkg/apis/app/v1beta1/status.go

```diff
@@ -82,6 +99,94 @@ func stsStatus(rsrc *appsv1.StatefulSet) string {
 	return StatusInProgress
 }
 
+// Deployment
+func deploymentStatus(rsrc *appsv1.Deployment) string {
+	status := StatusInProgress
+	progress := true
+	available := true
+	for _, c := range rsrc.Status.Conditions {
+		switch c.Type {
+		case appsv1.DeploymentProgressing:
+			// https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L52
+			if c.Status != corev1.ConditionTrue || c.Reason != "NewReplicaSetAvailable" {
+				progress = false
+			}
+		case appsv1.DeploymentAvailable:
+			if c.Status == corev1.ConditionFalse {
+				available = false
+			}
+		}
+	}
+
+	if progress && available {
```

Comment:
Can you help explain a bit more about the `progress` bool and the for loop above? Shouldn't this condition be `!progress && available`?

Reply:
Progressing is a continuous status here.
If it is not progressing it is due to an error.
It is not an indication of upgrade in progress.

https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/controller/deployment/progress.go#L85